### DISCUSSION TITLE
INI to JSON script

### DISCUSSION THREAD

**Discussion by lynshi** [microsoft/CCF#3700] [Q&A]
On [Line 111 in migrate_1_x_config.py](https://github.com/microsoft/CCF/blob/49d7b7941ea93e105c0e3a5bd542d94a0b75dd79/python/ccf/migrate_1_x_config.py#L111), there's the following:
```
output["command"]["start"]["constitution_files"] = json.loads(v)
```

Is this correct? I don't think the INI file expected JSON for the constitution argument, and it was previously just a file path (e.g. `constitution = constitution.js`). That wouldn't be JSON-parseable and so the migration script throws `json.decoder.JSONDecodeError`.

**Comment by jumaffre** [comment]
This is the 1.x sample `.ini` file that we use to check that the `migrate_1_x_config.py` script works: https://github.com/microsoft/CCF/blob/main/python/config_1_x.ini. 

As you can see, the existing `constitution` is expected to be a list of strings, which is valid JSON. What is the value of `constitution` in your `1.x` configuration file?

**Reply by lynshi** [reply]
This is what our start configuration looks like.
```
[start]
constitution = /path/to/constitution.js
network-cert-file = /path/to/networkcert.pem
member-info = "/path/to/public_key.pem,/path/to/keyshare.pem,/path/to/member_data.json"
```

I think the INI format was flexible enough to handle just a single string if there's no list of strings provided.

**Reply by lynshi** [reply]
As discussed offline, this is not a blocker for us, but thought it'd be useful to point out in case someone runs across this.
